hathor/consensus/block_consensus.py

```diff
@@ -289,28 +290,30 @@ def update_score_and_mark_as_the_best_chain_if_possible(self, block: Block) -> N
         self.update_score_and_mark_as_the_best_chain(block)
         self.remove_voided_by_from_chain(block)
 
+        best_score: int
         if self.update_voided_by_from_parents(block):
             storage = block.storage
             heads = [cast(Block, storage.get_transaction(h)) for h in storage.get_best_block_tips()]
-            best_score = 0.0
+            best_score = 0
             best_heads: list[Block]
             for head in heads:
                 head_meta = head.get_metadata(force_reload=True)
-                if head_meta.score <= best_score - self._settings.WEIGHT_TOL:
+                if head_meta.score < best_score:
```

Comment:
Why was the `=` removed?

Reply:
This loop is selecting the heads with highest score. As it's possible to have more than one head with the same score, it returns a list of blocks. This comparison is skipping heads that have lower score so they will no be selected at all. So we have to check if `head_meta.score < best_score`.

@glevco and @jansegre , confirm that you agree that I did the right thing. I'll wait for your confirmation before resolving this conversation.

Reply:
I agree.

Reply:
Resolving this, then.